cloud/deployment/deployment.go

```diff
@@ -719,28 +770,319 @@ func Update(deploymentID, label, ws, description, deploymentName, dagDeploy, exe
 				"\nRun `astro deploy` after this command to restart your DAGs. It may take a few minutes for the Airflow UI to update.")
 			confirmWithUser = true
 		}
-		deploymentUpdate.DagDeployEnabled = false
+		dagDeployEnabled = false
 	}
 
-	// if we have worker queues add them to the input
-	if len(wQueueList) > 0 {
+	configOption, err := GetDeploymentOptions("", astrocore.GetDeploymentOptionsParams{}, coreClient)
+	if err != nil {
+		return err
+	}
+
+	// build query input
+	updateDeploymentRequest := astroplatformcore.UpdateDeploymentRequest{}
+
+	if name == "" {
+		name = currentDeployment.Name
+	}
+	if description == "" {
+		if currentDeployment.Description != nil {
+			description = *currentDeployment.Description
+		}
+	}
+	var deploymentEnvironmentVariablesRequest []astroplatformcore.DeploymentEnvironmentVariableRequest
+	if len(newEnvironmentVariables) == 0 {
+		if currentDeployment.EnvironmentVariables != nil {
+			EnvironmentVariables := *currentDeployment.EnvironmentVariables
+			for i := range *currentDeployment.EnvironmentVariables {
+				var deploymentEnvironmentVariableRequest astroplatformcore.DeploymentEnvironmentVariableRequest
+				deploymentEnvironmentVariableRequest.IsSecret = EnvironmentVariables[i].IsSecret
+				deploymentEnvironmentVariableRequest.Key = EnvironmentVariables[i].Key
+				deploymentEnvironmentVariableRequest.Value = EnvironmentVariables[i].Value
+				deploymentEnvironmentVariablesRequest = append(deploymentEnvironmentVariablesRequest, deploymentEnvironmentVariableRequest)
```

Comment:
would this work as intended @fredzhy?
I'm worried we may replace secret env vars with `""`

Reply:
hmm interesting we could figure it out with testing

Reply:
yep we'll need to test this quite extensively
cc @shri-astro @sreenusuuda 

here is the core business logic comment for reference 
`apps/core/services/platform/shared/deployments/deployment_utils.go`
```
	// iterate through new variables and compose rollbackSecretEnvVars and secretEnvVarsToMutate
	// UPDATE PATH (if new variable is in the old variables):
	//   - if newVar.IsSecret && oldVar.IsSecret:
	//      * check if the new variable value is nil
	//      	> if value is nil, then that means there is no change to this variable, add it directly to the dbEnvVars
	//      	> if value is not nil,
	//        	a. read the old secret value out of vault, compare it with the new value
	//        	   i. if the same, then that means there is no change to this variable, add old variable to the dbEnvVars
	//					   ii. otherwise, update the updatedAt of the new variable, then add it to the dbEnvVars
	//        			   add the new variable to secretEnvVarsToMutate.SecretsToUpdate, then add the corresponding old variable to rollbackSecretEnvVars.SecretsToRevert
	//   - if newVar.IsSecret && !oldVar.IsSecret:
	//      * this means the new variable is a new secret, check if the value is nil
	//      	> if value is nil, return 400 error because a new secret variable cannot be nil
	//      	> if value is not nil,
	//        	a. update the updatedAt of the new variable, add it to the dbEnvVars
	//     			b. add the new variable to secretEnvVarsToMutate.SecretsToAdd and rollbackSecretEnvVars.SecretsToDelete
	//   - if !newVar.IsSecret && oldVar.IsSecret:
	//      * update the updatedAt of the new variable, add the new variable to the dbEnvVars,
	//      * read the secret value of the old variable out of vault,
	//      * add the old variable to secretEnvVarsToMutate.SecretsToDelete and rollbackSecretEnvVars.SecretsToAdd
	//   - if !newVar.IsSecret && !oldVar.IsSecret:
	//      * check if the old value and new value are the same
	//     		> if the same, add the old variable to the dbEnvVars
	//    	  > if not the same, update the updatedAt, add the new variable to the dbEnvVars
	// CREATE PATH (if new variable is not in the old variables):
	//    check if new variable is nil
	//    - if new variable is nil, return 400 error because a new variable cannot be nil
	//    - if new variable is not nil, check if it is a secret variable
	//     	* if new variable is a secret variable, add it to the dbEnvVars, secretEnvVarsToMutate.SecretsToAdd and rollbackSecretEnvVars.SecretsToDelete
	//      * if new variable is not a secret variable, add it to the dbEnvVars directly
	// DELETE PATH (if old variable is not in the new variables):
	//    check if old variable is a secret variable
	//    - if old variable is a secret variable, read the value out from vault, add it to secretEnvVarsToMutate.SecretsToDelete and rollbackSecretEnvVars.SecretsToAdd
```

Reply:
adding tests to bug bash